sdk/src/asset_handlers/bmff_io.rs

```diff
@@ -1547,6 +1556,268 @@ impl RemoteRefEmbed for BmffIO {
         Err(Error::UnsupportedType)
     }
 }
+
+#[allow(dead_code)]
+pub(crate) fn is_fragmented(asset: &mut dyn CAIRead) -> Result<bool> {
+    asset.rewind()?;
+
+    let start = asset.stream_position()?;
+    let size = asset.seek(SeekFrom::End(0))?;
+    asset.seek(SeekFrom::Start(start))?;
+
+    // create root node
+    let root_box = BoxInfo {
+        path: "".to_string(),
+        offset: 0,
+        size,
+        box_type: BoxType::Empty,
+        parent: None,
+        user_type: None,
+        version: None,
+        flags: None,
+    };
+
+    let (mut bmff_tree, root_token) = Arena::with_data(root_box);
+    let mut bmff_map: HashMap<String, Vec<Token>> = HashMap::new();
+
+    // build layout of the BMFF structure
+    build_bmff_tree(asset, size, &mut bmff_tree, &root_token, &mut bmff_map)?;
+
+    Ok(bmff_map.get("/moovf").is_some())
+}
+
+pub(crate) fn get_init_segment_boxes(asset: &mut dyn CAIRead) -> Result<Vec<BoxInfo>> {
+    asset.rewind()?;
+
+    let start = asset.stream_position()?;
+    let size = asset.seek(SeekFrom::End(0))?;
+    asset.seek(SeekFrom::Start(start))?;
+
+    // create root node
+    let root_box = BoxInfo {
+        path: "".to_string(),
+        offset: 0,
+        size,
+        box_type: BoxType::Empty,
+        parent: None,
+        user_type: None,
+        version: None,
+        flags: None,
+    };
+
+    let (mut bmff_tree, root_token) = Arena::with_data(root_box);
+    let mut bmff_map: HashMap<String, Vec<Token>> = HashMap::new();
+
+    // build layout of the BMFF structure
+    build_bmff_tree(asset, size, &mut bmff_tree, &root_token, &mut bmff_map)?;
+
+    let mut boxes = Vec::new();
+
+    if let Some(ftyp_tokens) = bmff_map.get("/ftyp") {
+        boxes.push(bmff_tree[ftyp_tokens[0]].data.clone());
+    }
+
+    if let Some(moov_tokens) = bmff_map.get("/moov") {
+        boxes.push(bmff_tree[moov_tokens[0]].data.clone());
+    }
+
+    Ok(boxes)
+}
+
+#[allow(dead_code)]
+struct SampleSizeBox {
```

Comment:
This doesn't look like it is exported anywhere, can you add a comment on why we are keeping it?

Reply:
It can go, I switched to use MP4 crate for this

Reply:
removed